### PR TITLE
fix(sqs-consumer.js): support sqs-consumer with a different signature

### PR DESCRIPTION
**Pull request by ranrib**
To support https://github.com/DaPulse/sqs-consumer

### CODE REVIEW

File: src/wrappers/sqs-consumer.js [hunk 1]
```diff
@@ -153,9 +159,9 @@ function sqsConsumerWrapper(wrappedFunction) {
         utils.debugLog('sqs-consumer - inside wrapper');
         utils.debugLog(`sqs-consumer - options: ${options}`);
         const app = wrappedFunction.apply(this, [options]);
-        const patchedCallback = message => traceContext.RunInContext(
+        const patchedCallback = (message, consumer) => traceContext.RunInContext(
             tracer.createTracer,
-            () => sqsConsumerMiddleware(message, app)
+            () => sqsConsumerMiddleware(message, consumer, app)
```
Comment (haimrait) [inline src/wrappers/sqs-consumer.js:164]: is the consumer  argument is mandatory?

File: src/wrappers/sqs-consumer.js [hunk 2]
```diff
@@ -40,14 +40,18 @@ function parseQueueUrl(queueUrl) {
 /**
  * Handle consumer event from sqs
  * @param {Array/SQSMessage} messages list of SQSMessage or a single one.
+ * @param {object} consumer object which sometimes can pass as an argument
  * @param {object} app consumer app.
+ * @return {object/Promise} Original handler result.
  */
-function sqsConsumerMiddleware(messages, app) {
+function sqsConsumerMiddleware(messages, consumer, app) {
     utils.debugLog('sqs-consumer - starting middleware');
```
Comment (haimrait) [inline src/wrappers/sqs-consumer.js:48]: is consumer is mandatory?

File: src/wrappers/sqs-consumer.js (hunk 1 above)
Comment (ranrib) [reply src/wrappers/sqs-consumer.js:164]: yes, it will be undefined if it's not sent

File: src/wrappers/sqs-consumer.js (hunk 2 above)
Comment (ranrib) [reply src/wrappers/sqs-consumer.js:48]: yes

**Review by haimrait — APPROVED** [review]

**Pull request merged**